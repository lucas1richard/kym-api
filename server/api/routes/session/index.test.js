const { expect, use } = require('chai');
const { jestSnapshotPlugin } = require('mocha-chai-jest-snapshot');

describe('session routes', () => {
  use(jestSnapshotPlugin());
  const routePrefix = '/api/session';

  describe(`GET ${routePrefix}/token/v1`, () => {
    const route = `${routePrefix}/token/v1`;
    before(globals.seedTestData);
    after(globals.destroyAllHook);

    it('gets user from token', async () => {
      const res = await globals.agent.getWithToken(route);
      expect(res.statusCode).equal(200);
      expect(res.body).toMatchSnapshot();
    });

    it('handles errors', async () => {
      const res = await globals.agent.get(route);
      expect(res.statusCode).equal(500);
    });
  });
  describe(`POST ${routePrefix}/signin/v1`, () => {
    const route = `${routePrefix}/signin/v1`;
    before(globals.seedTestData);
    after(globals.destroyAllHook);

    it('signs in and sends token', async () => {
      const { email, password } = globals.testData.users[0];
      const res = await globals.agent.post(route)
        .send({ email, password });
      expect(res.statusCode).equal(200);
      expect(res.body).toMatchSnapshot();
    });

    it('handles errors', async () => {
      const { email } = globals.testData.users[0];
      const res = await globals.agent.post(route)
        .send({ email, password: 'mockPassword' });
      expect(res.statusCode).equal(401);
    });
  });
  describe(`POST ${routePrefix}/signup/v1`, () => {
    const route = `${routePrefix}/signup/v1`;
    const userMeasurements = {
      gender: 'MALE',
      goal: 'LOSE_FAT',
      height: '73',
      date: '2018-01-01',
      lifestyle: 'NORMAL',
      units: 'IMPERIAL',
      weight: '179',
    };
    const body = {
      birthdate: '1989-01-01',
      email: 'test@test.test',
      firstname: 'Test',
      lastname: 'Test',
      password: '1234',
      preferredlocale: 'en-US',
      loggedIn: false,
    };
    beforeEach(globals.destroyAllHook);
    after(globals.destroyAllHook);
    it('signs up successfully', async () => {
      const res = await globals.agent.post(route)
        .send(body);
      expect(res.statusCode).equal(201);
    });
    it('signs up successfully with measurements', async () => {
      const res = await globals.agent.post(route)
        .send({
          ...body,
          userMeasurements,
        });
      expect(res.statusCode).equal(201);
    });
    it('fails with duplicate email', async () => {
      await globals.agent.post(route).send(body).expect(201);
      await agent.post(route).send(body).expect(400);
    });
    it('fails with duplicate email with measurements', async () => {
      await globals.agent.post(route).send({ ...body, userMeasurements }).expect(201);
      await globals.agent.post(route).send({ ...body, userMeasurements }).expect(400);
    });
    it('fails when there\'s a bad request', async () => {
      await globals.agent.post(route)
        .send({ ...body, firstname: undefined })
        .expect(400);
    });
  });
});
