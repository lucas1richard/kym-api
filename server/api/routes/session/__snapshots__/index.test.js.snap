// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`session routes GET /api/session/token/v1 gets user from token 1`] = `
Object {
  "birthdate": "1989-01-01",
  "email": "test@test.test",
  "firstname": "Test",
  "lastname": "Test",
  "preferredlocale": null,
  "username": "",
  "uuid": "60f24265-ef45-4891-b4ef-e8fad4435308",
}
`;

exports[`session routes POST /api/session/signin/v1 signs in and sends token 1`] = `
Object {
  "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.IjYwZjI0MjY1LWVmNDUtNDg5MS1iNGVmLWU4ZmFkNDQzNTMwOCI.LuRpYNcpwae3lJSHMfB727LewwjRb9Kb7Gh8v2BGsLw",
}
`;
